### DISCUSSION TITLE
Custom Product Sorting within Collections & Categories

### DISCUSSION THREAD

**Discussion by dfrnoch** [medusajs/medusa#3721] [Feature Requests]
In the current implementation, there isn't an option to manually arrange the products within a collection or category according to a store owner's preferences. This can make it difficult for store owners to highlight specific products or create a particular arrangement for a more visually appealing storefront.

I've recently come across a feature in Sylius that allows store owners to manually sort products within a collection using a "position" attribute. This approach could serve as inspiration for implementing a similar functionality in Medusa.

**Comment by riqwan** [comment]
Hey @lnxcz , we don't support custom ordering with collections yet, but categories do come with a "rank" column (which is our "position" attribute) that you can use to order based on your preferences - https://docs.medusajs.com/api/admin#tag/Product-Categories/operation/GetProductCategoriesCategory

**Reply by alsherif-khalaf** [reply]
I think custom ordering for products like the same way of ordering categories visually will be great feature and it is critical for store owners 
![image](https://github.com/medusajs/medusa/assets/73169053/69a8fe5a-44d2-4b77-9d1e-9fa7e63808b0)

**Comment by flashjames** [comment]
@lnxcz 
I implemented sorting and filtering directly with algolia with their Algolia Search product.
- Yes, the Search offering has all types of ways to categorize and filter products. I was a bit misled by the name Search and there's not any docs explaining the sorting and filtering use-case.
-  And the benefit is that I could just turn on their AI sorting product Algolia Recommend, that tunes sorting to each customer

If medusa had some route to add sorting with some type of rank on /store/products endpoint, I would have used that as start point.

If you @lnxcz or anybody else come's here and would like to know more details on this, send me an email at james@leky.hr for faster response since I don't watch github notifications every day.

@riqwan What you mentioned doesn't sort the products in the actual category what I can see?

**Reply by riqwan** [reply]
thanks for catching that @flashjames, I read the question incorrectly. You're right. We don't offer support for products ordering within a category or collection currently.

cc @lnxcz

**Reply by tekstackpk** [reply]
> thanks for catching that @flashjames, I read the question incorrectly. You're right. We don't offer support for products ordering within a category or collection currently.
> 
> cc @lnxcz

Medusa 2.0.0 is recently released. Can anyone please guide whether custom sorting of products in collections and categories is added as a feature or not?

**Comment by hamudeshahin** [comment]
@riqwan 
What about ui component, I am now working on implementing Mega menu for the app and I want to make customer be able to sorting the mega menu items just like you have did in the category ranking,
How could I take this component and use it in my custom page in Medusa v2??

**Comment by plenge** [comment]
We have several clients who are requesting this functionality.
We have made a temporary solution that works, but it would be very nice to see this be implemented natively in the Medusa admin.

Example image:
![Screenshot 2025-03-31 at 15 45 53](https://github.com/user-attachments/assets/6448b6d3-9e31-4103-965a-1b5e90ab6506)
